meshio/avsucd/_avsucd.py

```diff
@@ -212,59 +218,28 @@ def _write_cells(f, cells, cell_data, num_cells):
 
     # Material array
     if mat_data:
-        material = numpy.concatenate([v for v in cell_data[mat_data].values()])
+        material = numpy.concatenate(cell_data[mat_data])
     else:
         material = numpy.zeros(num_cells, dtype=int)
 
     # Loop over cells
     i = 0
     for k, v in cells:
-        for cell in v:
+        for cell in v[:, meshio_to_avsucd_order[k]]:
             cell_str = " ".join(str(c + 1) for c in cell)
             f.write(f"{i+1} {int(material[i])} {meshio_to_avsucd_type[k]} {cell_str}\n")
             i += 1
 
 
-def _write_node_data(f, point_data, num_nodes, num_node_data, num_node_data_sum):
-    num_node_data_str = " ".join(str(i) for i in num_node_data)
-    f.write(f"{len(num_node_data)} {num_node_data_str}\n")
+def _write_data(f, labels, data_array, num_entities, num_data, num_data_sum):
+    num_data_str = " ".join(str(i) for i in num_data)
+    f.write(f"{len(num_data)} {num_data_str}\n")
 
-    for label in point_data.keys():
+    for label in labels:
         f.write(f"{label}, real\n")
 
-    point_data_array = numpy.hstack(
-        [v[:, None] if v.ndim == 1 else v for v in point_data.values()]
-    )
-    point_data_array = numpy.hstack(
-        (numpy.arange(1, num_nodes + 1)[:, None], point_data_array)
-    )
-    numpy.savetxt(
-        f, point_data_array, delimiter=" ", fmt=["%d"] + ["%.14e"] * num_node_data_sum
-    )
-
-
-def _write_cell_data(f, cell_data, num_cells, num_cell_data, num_cell_data_sum):
-    num_cell_data_str = " ".join(str(i) for i in num_cell_data)
-    f.write(f"{len(num_cell_data)} {num_cell_data_str}\n")
-
-    for label in cell_data.keys():
-        if label not in meshio_data:
-            f.write(f"{label}, real\n")
-
-    cell_data_array = numpy.hstack(
-        [
-            vv[:, None] if vv.ndim == 1 else vv
-            for k, v in cell_data.items()
-            for vv in v.values()
-            if k not in meshio_data
-        ]
-    )
-    cell_data_array = numpy.hstack(
-        (numpy.arange(1, num_cells + 1)[:, None], cell_data_array)
-    )
-    numpy.savetxt(
-        f, cell_data_array, delimiter=" ", fmt=["%d"] + ["%.14e"] * num_cell_data_sum
-    )
+    data_array = numpy.column_stack((numpy.arange(1, num_entities + 1), data_array))
+    numpy.savetxt(f, data_array, delimiter=" ", fmt=["%d"] + ["%.14e"] * num_data_sum)
```

Comment:
savetxt is mega slow, perhaps we can replace this by `tofile` at some point in the future.